cryptographic_estimators/SDEstimator/SDAlgorithms/bjmm.py

```diff
@@ -405,100 +382,119 @@ def initialize_parameter_ranges(self):
 
+        Returns:
+            True if the parameter set is invalid, False otherwise.
         """
+        pass
         n, k, w = self.problem.get_parameters()
         par = SimpleNamespace(**parameters)
         k1 = (k + par.l) // 2
-        if par.p > w // 2 or k1 < par.p or par.l >= n - k or n - k - par.l < w - 2 * par.p or \
-                k1 - par.p < par.p2 - par.p / 2 or par.p2 < par.p // 2 or \
-                k1 - par.p2 < par.p1 - par.p2 / 2 or par.p1 < par.p2 / 2 or par.p % 2 == 1:
+        if (
+            par.p > w // 2
+            or k1 < par.p
+            or par.l >= n - k
+            or n - k - par.l < w - 2 * par.p
+            or k1 - par.p < par.p2 - par.p / 2
+            or par.p2 < par.p // 2
+            or k1 - par.p2 < par.p1 - par.p2 / 2
+            or par.p1 < par.p2 / 2
+            or par.p % 2 == 1
+        ):
             return True
         return False
 
     def _valid_choices(self):
         """
         Generator which yields on each call a new set of valid parameters based on the `_parameter_ranges` and already
-        set parameters in `_optimal_parameters`
-
+        set parameters in `_optimal_parameters`.
         """
         new_ranges = self._fix_ranges_for_already_set_parameters()
         n, k, w = self.problem.get_parameters()
 
         for p in range(new_ranges["p"]["min"], min(w // 2, new_ranges["p"]["max"]), 2):
             for l in range(new_ranges["l"]["min"], min(n - k - (w - 2 * p), new_ranges["l"]["max"])):
-                for p2 in range(max(new_ranges["p2"]["min"], p // 2 + ((p // 2) % 2)), new_ranges["p2"]["max"], 2):
-                    for p1 in range(max(new_ranges["p1"]["min"], (p2 + 1) // 2), new_ranges["p1"]["max"]):
-                        indices = {"p": p, "p1": p1, "p2": p2,
-                                   "l": l, "r": self._optimal_parameters["r"]}
+                for p2 in range(
+                    max(new_ranges["p2"]["min"], p // 2 + ((p // 2) % 2)),
+                    new_ranges["p2"]["max"],
+                    2,
+                ):
+                    for p1 in range(
+                        max(new_ranges["p1"]["min"], (p2 + 1) // 2),
+                        new_ranges["p1"]["max"],
+                    ):
+                        indices = {
+                            "p": p,
+                            "p1": p1,
+                            "p2": p2,
+                            "l": l,
+                            "r": self._optimal_parameters["r"],
+                        }
                         if self._are_parameters_invalid(indices):
                             continue
                         yield indices
 
     def _time_and_memory_complexity(self, parameters: dict, verbose_information=None):
-        """
-        computes the expected runtime and memory consumption for the depth 3 version
-        """
+        """Computes the expected runtime and memory consumption for the depth 3 version."""
```

Comment:
Should we alway allow new line before and after the each ` """ `?
So `"""Computes the expected runtime and memory consumption for the depth 3 version."""` becomes
```
"""
Computes the expected runtime and memory consumption for the depth 3 version.
"""
```

Reply:
I assumed that this is part of the google styling guide. If not I also prefer the newlines. But I would also make it dependent on if there is any format requirement by the doc generation by sphincs

Reply:
@Dioprz I don't understand why future editors have to fix from X to Y, X is what we want, or did you mean Y to X because of some automatic formatting that is applied?